framework/mongo/utils.py

```diff
@@ -60,16 +61,78 @@ def wrapper(cls):
         return cls
     return wrapper
 
-
-def get_or_http_error(Model, pk, allow_deleted=False):
-    instance = Model.load(pk)
+def get_or_http_error(Model, pk_or_query, allow_deleted=False, display_name=None):
+    """Load an instance of Model by primary key or modularodm.Q query. Raise an appropriate
+    HTTPError if no record is found or if the query fails to find a unique record
+    :param type Model: StoredObject subclass to query
+    :param pk_or_query:
+    :type pk_or_query: either
```

Comment:
Agreed; hence `type` is rather too broad for the argument expected here. If the resulting hint is too ambiguous, another option is to omit it entirely.